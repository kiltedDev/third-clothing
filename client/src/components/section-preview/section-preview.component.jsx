import React from 'react';

import SectionItem from '../section-item/section-item.component';

import {
  SectionPreviewContainer,
  SectionPreviewTitle,
  SectionPreviewPanel
} from './section-preview.styles';

export const SectionPreview = ({ title, items, linkURL }) => (
  <SectionPreviewContainer className='section-preview'>
    <SectionPreviewTitle
      className='title'
      to={ linkURL }
      children={ title }
    />
    <SectionPreviewPanel className='preview'>
      { items
        .filter((item, idx) => idx < 4)
        .map(( item ) => (
          <SectionItem key={item.id} item={ item } />
        ))
      }
    </SectionPreviewPanel>
  </SectionPreviewContainer>
);

export default SectionPreview;
