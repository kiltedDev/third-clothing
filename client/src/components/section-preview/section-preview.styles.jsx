import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const SectionPreviewContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;
  @media screen and (max-width: 800px) {
    align-items: center;
  }
`

export const SectionPreviewTitle = styled( Link )`
font-size: 28px;
margin-bottom: 25px;
font-weight: 600;
text-transform: uppercase;
@media screen and ( min-width: 801px ) {
  &:hover {
    color: gray;
    cursor: pointer;
  }
}
`

export const SectionPreviewPanel = styled.div`
  display: flex;
  justify-content: space-between;
  @media screen and (max-width: 800px) {
    display: grid;
    grid-template-columns: repeat( 4, 1fr );
    grid-gap: 15px;
    .menu-item, .section-item {
      grid-column: span 2;
      &:nth-last-child(1):nth-child(odd) {
        grid-column: 2 / span 2;
      }
    }
  }
`

SectionPreviewPanel.displayName = 'SectionPreviewPanel';
