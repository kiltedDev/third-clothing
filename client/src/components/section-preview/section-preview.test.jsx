import { shallow } from 'enzyme';
import React from 'react';
import { SectionPreview } from './section-preview.component';

describe('SectionPreview component', () => {
  let wrapper;
  let snapshot;
  const linkUrl = '/hats';
  const items = [
    {
      id: 1,
      name: 'Witch Hat'
    },
    {
      id: 2,
      name: 'Top Hat'
    },
    {
      id: 3,
      name: 'Fascinator'
    },
    {
      id: 4,
      name: 'Biggin'
    },
    {
      id: 5,
      name: 'Pefectly Acceptable Muggle Hat'
    },
  ];
  beforeAll(() => {
    const mockProps = {
      title: 'hats',
      items: items,
      linkURL: linkUrl
    };
    wrapper = shallow(<SectionPreview { ...mockProps } />);
    snapshot = wrapper.debug();
  });

  it('should render SectionPreview component',() => {
    expect( snapshot ).toMatchSnapshot();
  });

  it('should show no more than 4 items from a section',() => {
    expect( wrapper.find( 'SectionPreviewPanel' ).children() ).toHaveLength(4);
  });
});
