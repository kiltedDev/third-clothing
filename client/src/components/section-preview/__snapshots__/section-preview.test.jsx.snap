// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionPreview component should render SectionPreview component 1`] = `
"<styled.div className=\\"section-preview\\">
  <Styled(Link) className=\\"title\\" to=\\"/hats\\">
    hats
  </Styled(Link)>
  <SectionPreviewPanel className=\\"preview\\">
    <Connect(SectionItem) item={{...}} />
    <Connect(SectionItem) item={{...}} />
    <Connect(SectionItem) item={{...}} />
    <Connect(SectionItem) item={{...}} />
  </SectionPreviewPanel>
</styled.div>"
`;
